extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/mediaModels/GoogleMediaItem.java

```diff
@@ -107,7 +114,18 @@ public static PhotoModel convertToPhotoModel(
         albumId.orElse(null),
         false  /*inTempStore*/,
         null  /*sha1*/,
-        mediaItem.getUploadedTime());
+        getCreationTime(mediaItem));
+  }
+
+  private static Date getCreationTime(GoogleMediaItem mediaItem) throws Exception  {
+    // cannot be empty or null based. Verified the backend code.
+    try {
+      return CREATION_TIME_FORMAT.parse(mediaItem.getMediaMetadata().getCreationTime());
+    } catch (ParseException parseException) {
+      throw new ParseException(String.format("Skipping item transfer since code failed to parse the "
```

Comment:
Let's remove "skipping item transfer", since this is conjecture. We don't know what we're doing with this exception. What if this method is called in a new place that doesn't skip it. 

Also, if we're not creating a new Exception class for whatever error is thrown here, then there's no need to catch a ParseException and then throw a new ParseException, just an extra object being created.

Reply:
My understanding is this error would only be thrown whenever there's some kind of parsing issue which should happen only while interacting with metadata. In every case of corrupted metadata, we will be skipping the particular item so I feel okay keeping it here

Reply:
For this message to be accurate, every single caller of getCreationTime in the entire DTP codebase must be doing a transfer, and must be skipping those items, which is just not something you should be promising. 

The error message should tell us what the error is, not how it's going to be handled.

Reply:
Done